app/models/info_request.rb

```diff
@@ -239,6 +239,7 @@ def self.custom_states_loaded
   def self.guess_by_incoming_email(*emails)
     guesses = emails.flatten.reduce([]) do |memo, email|
       id, idhash = _extract_id_hash_from_email(email)
+      idhash ||= _guess_idhash_from_email(email)
```

Comment:
Though I haven't been able to reproduce this morning :(

Reply:
Ugh, this feels to me like it's somehow unintentionally matching `123ab` as `123` and returning a rogue result which is sometimes filtered out and sometimes not. Investigating.

Reply:
Oh, it's not even hitting the guess method - `_extract_id_hash_from_email` is doing that and select is sometimes pulling out the id match, sometimes the hash (I've commented out where we add the hash guess and it _always_ returns the `:id`). I'll change the test to prevent that.

Reply:
_sigh_

```
'123'.to_i
=> 123

'123ab'.to_i
 => 123

'ab123'.to_i
=> 0
```

Reply:
fixed in 01113a5